src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -994,13 +994,15 @@ class DataSetCache {
 };
 
 /** `{^ChooseFromDataset:{...}` */
-class RandomStringFromDataset : public Generator<std::string> {
+class ChooseStringFromDataset : public Generator<std::string> {
 
 public:
-    RandomStringFromDataset(const Node& node, GeneratorArgs generatorArgs)
+    ChooseStringFromDataset(const Node& node, GeneratorArgs generatorArgs)
         : _rng{generatorArgs.rng},
           _id{generatorArgs.actorId},
-          _path{node["path"].maybe<std::string>().value()} {
+          _path{node["path"].maybe<std::string>().value()},
+          _sequential(node["sequential"].maybe<bool>().value_or(false)),
+          _i(node["startFromLine"].maybe<uint64_t>().value_or(0)) {
```

Comment:
can we give this a better name than `_i`?

Reply:
Done.